lib/geoengineer/gps/gps.rb

```diff
@@ -156,6 +156,20 @@ def expanded_hash
 
   def loop_projects_hash(projects_hash, &block)
     projects_hash.each_pair do |project, environments|
+      # If the environments includes _other, pull out its value, and loop over
+      # all other known environments and add it back in, if it doesn't already
+      # exist. This allows _other to be used as a template for cookie cutter
+      # definitions.
+      if environments.key?("_other")
+        other = environments.delete("_other")
+
+        # NOTE: the constants appeared to be the best place to easily get all known environments.
+        all_environments = @constants.constants.keys.reject { |env| env.start_with?('_') }
+        all_environments.each do |env|
+          environments[env] = other.dup unless environments[env]
```

Comment:
Does `Hash.dup` perform a deep clone? Wondering if this could bite us in any way. Might not be a problem, I just get nervous when I see `dup`.

Reply:
We can use `HashUtils.deep_dup`... already defined helper function.